src/Motor.Extensions.Hosting.RabbitMQ/BasicPropertiesExtensions.cs

```diff
@@ -1,81 +1,89 @@
+        }
 
         public static void Update<T>(this IBasicProperties self, MotorCloudEvent<byte[]> cloudEvent,
-            RabbitMQPublisherOptions<T> options, ICloudEventFormatter cloudEventFormatter)
+            RabbitMQPublisherOptions<T> options)
         {
-            var messagePriority = cloudEvent.Extension<RabbitMQPriorityExtension>()?.Priority ??
-                                  options.DefaultPriority;
+            var messagePriority = cloudEvent.GetRabbitMQPriority() ?? options.DefaultPriority;
             if (messagePriority.HasValue)
                 self.Priority = messagePriority.Value;
-            var dictionary = new Dictionary<string, object>();
 
-            foreach (var attr in cloudEvent.GetAttributes())
+            var headers = new Dictionary<string, object>();
+
+            foreach (var (key, value) in cloudEvent.GetPopulatedAttributes())
             {
-                if (string.Equals(attr.Key, CloudEventAttributes.DataAttributeName(cloudEvent.SpecVersion))
-                    || string.Equals(attr.Key,
-                        CloudEventAttributes.DataContentTypeAttributeName(cloudEvent.SpecVersion))
-                    || string.Equals(attr.Key, RabbitMQPriorityExtension.PriorityAttributeName)
-                    || string.Equals(attr.Key, RabbitMQBindingExtension.ExchangeAttributeName)
-                    || string.Equals(attr.Key, RabbitMQBindingExtension.RoutingKeyAttributeName))
+                if (IgnoredAttributes.Contains(key))
+                {
                     continue;
-                dictionary.Add($"{CloudEventPrefix}{attr.Key}",
-                    cloudEventFormatter.EncodeAttribute(cloudEvent.SpecVersion, attr.Key, attr.Value,
-                        cloudEvent.GetExtensions().Values));
+                }
+
+                headers.Add($"{CloudEventPrefix}{key.Name}", Encoding.UTF8.GetBytes(key.Format(value)));
             }
 
-            self.Headers = dictionary;
+            self.Headers = headers;
         }
 
-        public static MotorCloudEvent<byte[]> ExtractCloudEvent<T>(this IBasicProperties self,
-            IApplicationNameService applicationNameService, ICloudEventFormatter cloudEventFormatter,
-            ReadOnlyMemory<byte> body,
-            IReadOnlyCollection<ICloudEventExtension> extensions)
+        public static MotorCloudEvent<byte[]> ExtractCloudEvent(this IBasicProperties self,
+            IApplicationNameService applicationNameService, ReadOnlyMemory<byte> body)
         {
-            var specVersion = CloudEventsSpecVersion.V1_0;
             var attributes = new Dictionary<string, object>();
             IDictionary<string, object> headers = new Dictionary<string, object>();
             if (self.IsHeadersPresent() && self.Headers is not null)
             {
                 headers = self.Headers;
             }
 
-            foreach (var header in headers
+            foreach (var (key, value) in headers
```

Comment:
nit: I would prefer to split filtering and looping of the headers because this is really hard to read :sweat_smile: 

suggestion:

```cs
            var cloudEventHeaders = headers
                .Where(t => t.Key.StartsWith(CloudEventPrefix))
                .Select(t => (Key: t.Key[CloudEventPrefix.Length..], t.Value))
                .Where(t => !t.Key.Equals(CloudEventsSpecVersion.SpecVersionAttribute.Name, StringComparison.InvariantCultureIgnoreCase));
            
            foreach (var (key, value) in cloudEventHeaders)
            {
                attributes.Add(key, value);
            }
```